server.go

```diff
@@ -824,6 +834,12 @@ func (s *Server) Shutdown() {
 	graceful.Shutdown()
 }
 
+// CalculateTickDelay takes the provided time, `Truncate`s it a rounded-down
+// multiple of `interval`, then adds `interval` back to find the "next" tick.
+func (s *Server) CalculateTickDelay(t time.Time) time.Duration {
```

Comment:
does this need to be a method on `Server`? We could just pass in the interval as a separate argument, which would make it easier to test without needing to spin up a new server.